src/components/ListItem.js

```diff
@@ -76,20 +75,87 @@ const ListItem = () => {
     setItems(items);
   };
 
+  const addBgColor = (nextPurchase) => {
+    if (nextPurchase === 7) {
+      return 'green-bg';
+    } else if (nextPurchase === 14) {
+      return 'yellow-bg';
+    } else {
+      return 'purple-bg';
+    }
+  };
+
+  const addAriaLabel = (nextPurchase, itemName) => {
+    if (itemName && nextPurchase === 7) {
+      return `Buy ${itemName} every ${nextPurchase}} days`;
+    } else if (itemName && nextPurchase === 14) {
+      return `Buy ${itemName} every ${nextPurchase}} days`;
+    } else if (itemName && nextPurchase === 30) {
+      return `Buy ${itemName} every ${nextPurchase}} days`;
+    }
+  };
+
+  const DaysInMilliSeconds = 60 * 60 * 24 * 1000;
+
+  const calculateActive = (item) => {
+    if (item.totalPurchases <= 1) {
+      return false;
+    } else {
+      return (
+        ((Date.now() - item.lastPurchase) / item.estimatedPurchaseInterval) *
+          DaysInMilliSeconds >
+        2
+      );
+    }
+  };
+
+  const sortItems = (items) => {
+    return items.sort((a, b) => {
+      if (!calculateActive) {
+        return -1;
+      }
+
+      if (!a.estimatedPurchaseInterval && !b.estimatedPurchaseInterval) {
+        if (a.itemName < b.itemName) {
+          return -1;
+        } else {
+          return 1;
+        }
+      }
+
+      if (a.estimatedPurchaseInterval < b.estimatedPurchaseInterval) {
+        return -1;
+      } else if (a.estimatedPurchaseInterval > b.estimatedPurchaseInterval) {
+        return 1;
+      } else {
+        if (a.itemName < b.itemName) {
+          return -1;
+        } else {
+          return 1;
+        }
```

Comment:
Since this code is repeated twice, this code could be in a function to avoid code duplication